src/Data/List/NonEmpty/Extra.hs

```diff
@@ -62,6 +64,29 @@ sortOn f = fromList . List.sortOn f . toList
 union :: Eq a => NonEmpty a -> NonEmpty a -> NonEmpty a
 union = unionBy (==)
 
+-- | @nubOrd@ for 'NonEmpty'. Behaves the same as 'Data.List.Extra.nubOrd'.
+--
+-- > Data.List.NonEmpty.Extra.nubOrd (1 :| [2, 3, 3, 4, 1, 2]) == 1 :| [2, 3, 4]
+-- > \xs -> Data.List.NonEmpty.Extra.nubOrd xs == Data.List.NonEmpty.Extra.nub xs
+nubOrd :: Ord a => NonEmpty a -> NonEmpty a
+nubOrd = nubOrdBy compare
+
+-- | @nubOrdBy@ for 'NonEmpty'. Behaves the same as 'Data.List.Extra.nubOrdBy'.
+--
+-- > Data.List.NonEmpty.Extra.nubOrdBy (compare `on` Data.List.length) ("a" :| ["test","of","this"]) == "a" :| ["test","of"]
+nubOrdBy :: (a -> a -> Ordering) -> NonEmpty a -> NonEmpty a
+nubOrdBy cmp = fromList . List.nubOrdBy cmp . toList
+
+-- | @nubOrdOn@ for 'NonEmpty'. Behaves the same as 'Data.List.Extra.nubOrdOn'.
+--
+-- > Data.List.NonEmpty.Extra.nubOrdOn Data.List.length ("a" :| ["test","of","this"]) == "a" :| ["test","of"]
+nubOrdOn :: Ord b => (a -> b) -> NonEmpty a -> NonEmpty a
+nubOrdOn f = fmap snd . nubOrdBy (compare `on` fst) . fmap (f &&& id)
+
+-- | @nubOn@ for 'NonEmpty'. Behaves the same as 'Data.List.Extra.nubOn'.
+nubOn :: Eq b => (a -> b) -> NonEmpty a -> NonEmpty a
+nubOn f = fmap snd . nubBy ((==) `on` fst) . fmap (\x -> let y = f x in y `seq` (y, x))
```

Comment:
I'd probably ditch this function. nubOn is O(n^2). I don't think adding it to Extra was a good idea, and I might deprecate it in future.